### DISCUSSION TITLE
Fails to discover entities with pnpm workspace

### DISCUSSION THREAD

**Discussion by fortizflexidao** [mikro-orm/mikro-orm#6046] [General]
### Describe the bug

Under a pnpm workspace set with `shared-workspace-lockfile=false`, there seems to be issues with the decorators. I tried to analyze the issue but I don't know how decorators work. I think that it might be related the code that scans the stack trace in `Utils.lookupPathFromDecorator`, but I cannot be certain. The error message is: 

```Only abstract entities were discovered, maybe you forgot to use @Entity() decorator?```

You can edit the `.npmrc` file and set `shared-workspace-lockfile` to `true`, clear all `node_modules`:

```find . -name node_modules -prune -exec rm -r {} \; ```

Run `pnpm i` and it will work.

__PS: When it does not fail I still get an error stating `Could not locate the bindings file`, but it is not related to the task.__

### Reproduction

https://github.com/fortizflexidao/mikro-orm-pnpm-bug-reproduction

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.3.10 and 6.3.11-dev.3

### Node.js version

20.17.0

### Operating system

MacOS + Linux

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
This is most likely caused by having multiple ORM packages installed, therefore having multiple global registers for the decorator metadata. You can't do that if you want to use decorators, there needs to be only one `@mikro-orm/core` package.

**Reply by fortizflexidao** [reply]
Thank you very much. This is the information I was lacking. I just re-exported `@mikro-orm/postgresql` from our database package `index.ts` and updated all the imports to point to the re-exported classes and now it works.
